File: SlidingWindow/1876_substrings_of_size_three/main.py
import collections


class Solution:
    def countGoodSubstrings(self, s: str) -> int:
        left, nums, state = 0, 0, collections.defaultdict(int)

        for idx, elem in enumerate(s):
            state[elem] += 1

            if idx - left == 3:
                state[s[left]] -= 1
                if state[s[left]] == 0:
                    del (state[s[left]])
                left += 1

            if len(state) == 3:
                nums += 1
        return nums

File: SlidingWindow/1876_substrings_of_size_three/main.go
package main

func countGoodSubstrings(s string) int {
	var left, nums int
	state := map[int32]int{}
	for idx, elem := range s {
		state[elem]++
		if idx-left == 3 {
			left_elem := int32(s[left])
			state[left_elem]--
			if state[left_elem] == 0 {
				delete(state, left_elem)
			}
			left++
		}
		if len(state) == 3 {
			nums++
		}
	}
	return nums
}
